embc-app/ClientApp/src/app/volunteer/components/valid-from-to/valid-from-to.component.html

```diff
@@ -1,36 +1,46 @@
 <div class="row">
-  <div class="col-8">
+  <div class="col">
     <span class="h3">Valid From</span>
     <div class="row">
-      <div class="col-6">
-        <label for="validFromDate">Date</label>
-        <input type="text" id="validFromDate" class="form-control" [value]="validFrom">
-        <span class="invalid-feedback">Please enter a valid date.</span>
+      <div class="col">
+        <div class="input-group mb-0">
+          <label for="validFromDate_{{id}}">Date</label>
+          <input class="form-control" placeholder="yyyy-mm-dd" name="validFromDate_{{id}}" ngbDatepicker #fromDate="ngbDatepicker" [firstDayOfWeek]="7" navigation="arrows" [(ngModel)]="wrdForm.fromDate" (click)="fromDate.close();" (ngModelChange)="updateDisplay()" required aria-required="true">
+          <div class="input-group-append">
+            <button class="btn btn-icon picker-patch" (click)="fromDate.toggle()" type="button" tabindex="-1">
+              <i class="material-icons">date_range</i>
+            </button>
+          </div>
+          <span *ngIf="!validDate" class="invalid-feedback" style="display: block;">Please enter a valid date.</span>
+        </div>
       </div>
-      <div class="col-3 pl-0">
-        <label for="validFromTime">Time</label>
-        <input type="time" id="validFromTime" class="form-control">
-        <span class="invalid-feedback">Please enter a valid time.</span>
+      <div class="col-3 pl-0 pr-0">
+        <div class="form-group mb-0">
+          <label for="validFromTime_{{id}}">Time</label>
```

Comment:
The 'for' here doesn't actually work (at least for clicking on the label and having the input focused. I don't know if this can be fixed easily so if you look into it then I suggest time-boxing it.

Reply:
Not easily fixable because the field is inside the ng bootstrap component. Will look later when we have time to bite into tech debt.